ic-stable-storage/src/map.rs

```diff
@@ -0,0 +1,232 @@
+use std::mem::size_of;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::rc::Rc;
+
+use candid::{CandidType, Deserialize};
+
+use super::error::Result;
+use super::{from_bytes, Memory, RestrictedMemory, StableBTreeMap, StableMemory, VirtualMemory, PADDING};
+
+type Mem<const INDEX: u8> = VirtualMemory<Rc<RestrictedMemory<StableMemory>>, INDEX>;
+
+/// An append only log.
+/// Inserting the same value twice will simply replace the inner value.
```

Comment:
Good spot!

Reply:
An example: In Neuro, we use the trait of IS20, some states are initialized in IS20, and then we need to initialize other states in Neuro.

If using names, when multiple traits are introduced, there is also a chance that different traits will use the same name and cause conflicts.

A possible solution is to leave a parameter in the trait, so that when using traits, devs can decide which index to use for the state in traits.

Reply:
This sounds like a hot mess that makes no sense.

If we look at this from the perspective of an end-user:
It is already a stretch to explain to the end user why they have to put some magic number into their type, and how that number can't be the same as another instance's  number.

Now they have to specify some kind of namespace on their trait?

We need a better solution than this.
I'll bring this up in chat so we can discuss there